Utilities/DavixAdaptor/src/DavixFile.cc

```diff
@@ -33,8 +33,9 @@ DavixFile::~DavixFile(void) {
 
 void DavixFile::close(void) {
   if (m_davixPosix && m_fd) {
+    auto davixPosix = std::move(m_davixPosix);
     DavixError *err = nullptr;
-    m_davixPosix->close(m_fd, &err);
+    davixPosix->close(m_fd, &err);
```

Comment:
How about also setting `m_fd` to `nullptr`?

```suggestion
    davixPosix->close(m_fd, &err);
    m_fd = nullptr;
```

Reply:
Sure, although to use `m_fd` one first has to dereference `m_davixPosix`.